src/masternodes/mn_checks.cpp

```diff
@@ -310,6 +320,23 @@ Res ApplyCreateTokenTx(CCustomCSView & mnview, CCoinsViewCache const & coins, CT
         }
     }
 
+    if (token.decimal > 18) {
+        return Res::Err("%s : %s", base, "Decimal is too deep");
+    }
+
+    CAmount coin = std::pow(10, token.decimal);
+    if (token.limit == 0) {
+        auto limit = MAX_MONEY / COIN;
```

Comment:
Here should be auto limit = MAX_MONEY / coin? coin and COIN very easy to get confused, i suggest to rename one of them.

Reply:
No, here is to determine what's the base value without precision.